### PR TITLE
feat(s3): add `s3_bucket_cross_region_replication` check

### PR DESCRIPTION
**Pull request by HugoPBrito**
### Context

General-purpose buckets in Amazon S3 should have cross-Region replication enabled to enhance data availability and resilience. 

Cross-Region replication involves automatically and asynchronously copying objects between buckets located in different AWS Regions. This ensures that newly created objects and updates in a source bucket are replicated to a destination bucket.

### Description

This new check called `s3_bucket_cross_region_replication` verifies whether cross-Region replication is configured, and flags any instances where it is not. 

If cross-Region replication is not needed for a specific bucket, exceptions can be made, and findings for that bucket can be suppressed.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [ ] Review if the code is being covered by tests.
- [ ] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/s3/s3_bucket_cross_region_replication/s3_bucket_cross_region_replication.metadata.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  "ResourceType": "AwsS3Bucket",
+  "Description": "Verifying whether S3 buckets have cross-region replication enabled, ensuring data redundancy and availability across multiple AWS regions",
+  "Risk": "Without cross-region replication in S3 buckets, data is at risk of being lost or inaccessible if an entire region goes down, leading to potential service disruptions and data unavailability.",
+  "RelatedUrl": "https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication.html",
+  "Remediation": {
+    "Code": {
+      "CLI": "",
+      "NativeIaC": "https://docs.prowler.com/checks/aws/general-policies/ensure-that-s3-bucket-has-cross-region-replication-enabled/",
+      "Other": "https://docs.aws.amazon.com/securityhub/latest/userguide/s3-controls.html#s3-7",
+      "Terraform": ""
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/s3/s3_bucket_cross_region_replication/s3_bucket_cross_region_replication.metadata.json]:
```suggestion
      "Terraform": "https://docs.prowler.com/checks/aws/general-policies/ensure-that-s3-bucket-has-cross-region-replication-enabled#terraform"
```

File: prowler/providers/aws/services/s3/s3_bucket_cross_region_replication/s3_bucket_cross_region_replication.metadata.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  "ResourceIdTemplate": "arn:partition:s3:::bucket_name",
+  "Severity": "low",
+  "ResourceType": "AwsS3Bucket",
+  "Description": "Verifying whether S3 buckets have cross-region replication enabled, ensuring data redundancy and availability across multiple AWS regions",
+  "Risk": "Without cross-region replication in S3 buckets, data is at risk of being lost or inaccessible if an entire region goes down, leading to potential service disruptions and data unavailability.",
+  "RelatedUrl": "https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication.html",
+  "Remediation": {
+    "Code": {
+      "CLI": "",
+      "NativeIaC": "https://docs.prowler.com/checks/aws/general-policies/ensure-that-s3-bucket-has-cross-region-replication-enabled/",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/s3/s3_bucket_cross_region_replication/s3_bucket_cross_region_replication.metadata.json]: This link does not have CloudFormation.
```suggestion
      "NativeIaC": "",
```

File: prowler/providers/aws/services/s3/s3_bucket_cross_region_replication/s3_bucket_cross_region_replication.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            report.resource_arn = arn
+            report.resource_tags = bucket.tags
+
+            if (
+                bucket.versioning
+                and bucket.replication
+                and bucket.replication.status == "Enabled"
+                and bucket.replication.destination
+            ):
+                destination_bucket = s3_client.buckets.get(
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/s3/s3_bucket_cross_region_replication/s3_bucket_cross_region_replication.py]: What if the bucket is in another account or if Prowler is only auditing one region?

File: prowler/providers/aws/services/s3/s3_service.py [hunk 4]
```diff
@@ -378,6 +379,37 @@ def _get_bucket_tagging(self, bucket):
...
+    def _get_bucket_replication(self, bucket):
+        logger.info("S3 - Get buckets replication...")
+        try:
+            regional_client = self.regional_clients[bucket.region]
+            replication_config = regional_client.get_bucket_replication(
+                Bucket=bucket.name
+            )["ReplicationConfiguration"]["Rules"]
+            if replication_config:
+                bucket.replication = Rule(
+                    status=replication_config[0]["Status"],
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/s3/s3_service.py]: The buckets can have more than 1 replication rule, change the check accordingly please.

File: prowler/providers/aws/services/s3/s3_bucket_cross_region_replication/s3_bucket_cross_region_replication.py (hunk 3 above)
Comment (MrCloudSec) [reply prowler/providers/aws/services/s3/s3_bucket_cross_region_replication/s3_bucket_cross_region_replication.py]: Solved with my commit.

**Comment by sergargar** [comment]
I improved the logic of the check by adding more use cases, please read the changes and add a test for each new use case. Let me know if you don't understand the changes. Thanks!

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
I improved the logic of the check by adding more use cases, please read the changes and add a test for each new use case. Let me know if you don't understand the changes. Thanks!

File: prowler/providers/aws/services/s3/s3_bucket_cross_region_replication/s3_bucket_cross_region_replication.py [hunk 5]
```diff
@@ -13,26 +13,26 @@ def execute(self):
...
-                        bucket.replication.destination
-                    ]
-                    if destination_bucket.region != bucket.region:
-                        report.status = "PASS"
-                        report.status_extended = f"S3 Bucket {bucket.name} has cross region replication in bucket {destination_bucket.name} located in region {destination_bucket.region}."
-                    else:
-                        report.status = "FAIL"
-                        report.status_extended = f"S3 Bucket {bucket.name} has cross region replication in bucket {destination_bucket.name} located in the same region."
-
+            if bucket.replication_rules:
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/s3/s3_bucket_cross_region_replication/s3_bucket_cross_region_replication.py:16]: Please, change the logic so one cross-region replication rule is enough for a bucket.

**Review by MrCloudSec — DISMISSED** [review]
Love this new check!

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**